### PR TITLE
fix(KFLUXBUGS-1786): git provider of gitlab.cee.redhat.com is gitlab

### PR DESCRIPTION
**Pull request by testcara**
## Fixes 
[KFLUXBUGS-1786](https://issues.redhat.com/browse/KFLUXBUGS-1786)

## Description
When people create component and specify the git source as 'https://gitlab.cee.redhat.com/test.git', the git provider should be auto-filled as 'gitlab'.

## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 
Screenshot 2024-10-30 at 17 53 55

## How to test or reproduce?
1. create component
2. input the https://gitlab.cee.redhat.com/test.git as the git repo url
3. click the 'show advanced git' to ensure the 'gitlab' is shown as the git provider annotation.

## Browser conformance: 

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: src/shared/utils/git-utils.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ export const detectGitType = (url: string): GitProvider => {
   if (hasDomain(url, 'bitbucket.org')) {
     return GitProvider.BITBUCKET;
   }
-  if (hasDomain(url, 'gitlab.com')) {
+  if (hasDomain(url, 'gitlab.com') || hasDomain(url, 'gitlab.cee.redhat.com')) {
```
Comment (MartinBasti) [inline src/shared/utils/git-utils.tsx:35]: Why internal redhat gitlab is being hardcoded in upstream codebase?

This should be a config if you want to have custom mapping

Comment (testcara) [reply src/shared/utils/git-utils.tsx:35]: Good point. You're right, the hardcode is a bit weird.
And I guess selecting git provider directly is much cost-effective than developing more other extra custom mapping here.
So I would confirm with the customer to close the PR directly.

**Comment by testcara** [comment]
FYI, after confirmed with the reporter, it is ok to close the PR directly. Thank you for your attention.

**Comment by testcara** [comment]
Close it now.

**Pull request closed without merging**